run.py

```diff
@@ -9,61 +9,61 @@
 from library.evaluation import classification_metrics
 
 from blocks.pipeline import Pipeline
-from typing import Tuple
+from typing import List, Tuple
 from plugins import WandbPlugin, WandbConfig
-from type import PreprocessConfig
+from type import PreprocessConfig, RunConfig
 
 
 hate_speech_data = load_data("data/original", preprocess_config)
 
 
 def run(
     pipeline: Pipeline,
-    data: Tuple[Dataset, Dataset],
     preprocess_config: PreprocessConfig,
     project_id: str,
+    run_configs: List[RunConfig],
 ) -> None:
-    train_dataset, test_dataset = data
 
-    train_runner = Runner(
-        pipeline,
-        data={"input": train_dataset[Const.input_col]},
-        labels=train_dataset[Const.label_col],
-        evaluators=classification_metrics,
-        train=True,
-        plugins=[
-            WandbPlugin(
-                WandbConfig(
-                    project_id=project_id,
-                    run_name=pipeline.id,
-                    train=True,
-                ),
-                dict(pipeline.get_configs(), preprocess_config=vars(preprocess_config)),
-            )
-        ],
-    )
-    train_runner.run()
-
-    test_runner = Runner(
-        pipeline,
-        data={"input": test_dataset[Const.input_col]},
-        labels=test_dataset[Const.label_col],
-        evaluators=classification_metrics,
-        train=False,
-        plugins=[
-            WandbPlugin(
-                WandbConfig(project_id=project_id, run_name=pipeline.id, train=False),
-                dict(pipeline.get_configs(), preprocess_config=vars(preprocess_config)),
-            )
-        ],
-    )
-    test_runner.run()
+    for config in run_configs:
+        runner = Runner(
+            config,
+            pipeline,
+            data={Const.input_col: config.dataset[Const.input_col]},
+            labels=config.dataset[Const.label_col]
+            if hasattr(config.dataset, Const.label_col)
+            else None,
+            evaluators=classification_metrics,
+            plugins=[
+                WandbPlugin(
```

Comment:
should we by default not log when `train` == True? I think its a specific debugging step, that should be off by default?

Reply:
I don't see why we should log when we are training! 
All projects I have seen log the training process to be able to catch any errors/monitor the process & resources. For small models it might not make a huge difference but for larger models its essential.

Reply:
ohh I see, I meant not logging accuracy metrics, as they're irrelevant.
but, you're right, for DL, we want to log training stuff (not f1, but loss over time, etc.), so let's do that!

Reply:
I'll think of a way to define what to log! huggingface definitely allows for that